apps/omg/lib/omg/state/transaction/payment.ex

```diff
@@ -118,9 +118,12 @@ defmodule OMG.State.Transaction.Payment do
          do: {:ok, outputs}
   end
 
-  defp reconstruct_metadata([]), do: {:ok, @zero_metadata}
-  defp reconstruct_metadata([metadata]) when Transaction.is_metadata(metadata), do: {:ok, metadata}
-  defp reconstruct_metadata([_]), do: {:error, :malformed_metadata}
+  # txData is required to be zero in the contract, zero comes out as `""` from `ExRLP.decode/1`
+  defp check_tx_data(""), do: :ok
```

Comment:
Since it's `uint256` we can check the value of `RawData.parse_uint256(tx_data) == 0` and hide the weirdness of RLP from here

Reply:
ah ok, I realized this earlier too and was thinking that your PR would help here :sweat_smile:. I'll try to do that if not too much hassle